src/util/arr.h

```diff
@@ -111,7 +111,7 @@ static inline uint32_t array_len(array_t arr) {
 static inline void *array_trimm(array_t arr, uint32_t len, uint32_t cap) {
   array_hdr_t *arr_hdr = array_hdr(arr);
   assert(len >= 0 && "trimming len is negative");
-  assert((cap == -1 || cap > 0) && "trimming capacity is illegal");
+  assert((cap == -1 || cap > 0 || len == cap) && "trimming capacity is illegal");
```

Comment:
Why the change here? It's perfectly legal in most APIs to trim with a 'zero' effect. I'd agree that I can't find a reason for that, but why the change?

Reply:
if cap is zero them len should also be zero, len can not be greater then cap.